client/src/templates/RouteWithSubRoutes.js

```diff
@@ -32,10 +32,19 @@ export const RouteWithSubRoutes = (route) => {
   return (
     <Route
       path={route.path}
-      render={(props) =>
-        // pass the sub-routes down to keep nesting
-        switchLayout(route.layout, { ...props, ...route.props })
-      }
+      render={(props) => {
+        return !props.isPrivate || props.isAuthenticated ? (
+          // pass the sub-routes down to keep nesting
+          switchLayout(route.layout, { ...props, ...route.props })
+        ) : (
+          <Redirect
```

Comment:
Good idea to have a redirect route. We could also generalize routes to be either `protected` or `auth` routes. So for instance, protected routes would be paths that require a loggedin user in order to access, otherwise it will redirect the user to login/signup.

Reply:
yeah that's what I was thinking, we could make components for that, and we could maybe set a prop on the definitions in the `routes` file... but currently it's a bit mixed into the whole `layout templates` file which is confusing, not sure if we need that and how we can improve that (maybe wrap multiple HOCs).

Reply:
Yea I've made route components before in personal projects and having those components are pretty handy for routing.

Reply:
Maybe we should make some more separate tickets to work on these things.